### PR TITLE
[Kubernetes Plugin] expose detectErrors function publicly

### PR DESCRIPTION
**Pull request by goenning**
Signed-off-by: goenning <me@goenning.net>

## Hey, I just made a Pull Request!

We often build aggregate views to help teams navigate through backstage when they own dozens of components. One of these views is built on top of the Kubernetes plugin, the challenge we have currently is that we want to make the errors consistent across the aggregate and single component view, exposing the detectErrors function would make this possible

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by goenning** [comment]
I didn’t know backstage had its own error detection functions until I started building this aggregate view. I always thought the list of errors was provided by the API.

But looking at the link you posted, it doesn’t seem to provide a similar list of errors, for instance, one that i like a lot is when a container restarts > X times, that’s super useful to have in backstage, but kstatus doesn’t seem to have that.

Either way, If this plug-in adopts a different strategy in future, it’ll be a breaking change and integrators who used this function will have to deal with it.

 I’d still like to use what we have today, we could also export it as experimentalDetectErrors to indicate that an API change is likely to happen in future.

File: plugins/kubernetes/api-report.md [hunk 1]
```diff
@@ -108,6 +108,38 @@ export interface DeploymentResources {
   replicaSets: V1ReplicaSet[];
 }
 
+// Warning: (ae-missing-release-tag) "DetectedError" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
```
Comment (Rugvip) [inline plugins/kubernetes/api-report.md:111]: Let's not introduce any new API report warnings, please handle these

**Review by Rugvip — COMMENTED** [review]
Feeling this could potentially be a Utility API that one can override to add custom error detection logic? Or if these errors are easily serializable and could be useful to enumerate from other sources, we could even move the detection to the backend and make it pluggable there?

What do you think about the existing change as it is @mclarke47, is it polluting the API and introducing too much of a maintenance burden, or does it make sense to expose?

File: plugins/kubernetes/api-report.md (hunk 1 above)
Comment (goenning) [reply plugins/kubernetes/api-report.md:111]: Sure :+1: I'll fix this once there's an agreement if this will go ahead as is or not.

**Comment by goenning** [comment]
I really like the idea of making it extensible too, it'd open so many possibilities 😁

**Comment by goenning** [comment]
Not stale.

**Comment by mclarke47** [comment]
@Rugvip It would be great to have a pluggable API as people might want to detect errors on closed source custom resources, perhaps we should move the error detection to the backend and make it more extensible

**Comment by mclarke47** [comment]
We can probably tidy this up and merge for now though

**Review by mclarke47 — APPROVED** [review]

**Comment by benjdlambert** [comment]
It's worth noting that in the current setup these imports will still be available under the `@backstage/plugin-kubernetes` import rather than `@backstage/plugin-kubernetes/alpha` I wonder if we should add the tooling required for the `/alpha` path before this? /cc @mclarke47 @Rugvip

**Comment by benjdlambert** [comment]
But actually, looking through the current `api-report` theres quite a lot of usage of it already, so enabling the alpha types would be a breaking change. But undecided on the best path forward here. The library isn't 1.x yet, so maybe doesn't need the `/alpha` import path as everything is kind of unstable anyways. Thoughts?

**Comment by benjdlambert** [comment]
Ok, think we have a path forward after chatting with @Rugvip, and I think we hold off on the `/alpha` tooling for now until things go v1.0 in this repo, and then we can move all the experimental stuff at that time under that import instead as there's a lot of stuff that is currently alpha and it would be a breaking change and it's not worth the hassle.

**Review by benjdlambert — APPROVED** [review]
Thanks :pray:

**Pull request merged**